aten/src/ATen/native/ConstantPadNd.cpp

```diff
@@ -0,0 +1,54 @@
+#include "ATen/ATen.h"
+
+namespace at { namespace native {
+
+Tensor constant_pad_nd(const Tensor& input, IntList pad, Scalar value) {
+    auto input_sizes = input.sizes();
+    auto l_inp = input.dim();
+
+    auto l_pad = pad.size() / 2;
+    auto l_diff = l_inp - l_pad;
+    AT_CHECK(l_inp >= l_pad, "Padding length too large");
+
+    std::vector<int64_t> new_shape;
+
+    for (int i = 0; i < l_diff; i ++) {
+        new_shape.push_back(input_sizes[i]);
+    }
+
+    for (int i = 0; i < l_pad; i++) {
+        auto pad_idx = pad.size() - ((i + 1) * 2);
+        auto new_dim = input_sizes[l_diff + i] + pad[pad_idx] + pad[pad_idx + 1];
+        AT_CHECK(new_dim > 0, "input is too small");
+        new_shape.push_back(new_dim);
+    }
+
+    auto output = at::empty(new_shape, input.options());
+    output.fill_(value);
+
+    auto c_input = input;
```

Comment:
You probably need to call `clone` on `input`.

Reply:
I don't think `clone` is necessary here, because there are no operations on `c_input` that are in-place (it calls `narrow` rather than `narrow_`).